### PR TITLE
raise ResourceNotFound on a 404

### PR DESCRIPTION
**Pull request by bdunne**
https://bugzilla.redhat.com/show_bug.cgi?id=1599754

### CODE REVIEW

**Comment by bdunne** [comment]
Found a second instance of https://github.com/ManageIQ/manageiq-api-client/pull/82

**Comment by bdunne** [comment]
@jrafanie Please review

**Comment by jrafanie** [comment]
@bdunne looks good but some tests failed with the now proper resource not found error

File: lib/manageiq/api/client/connection.rb [hunk 1]
```diff
@@ -103,7 +103,10 @@ def send_request(method, path, params, &block)
         end
 
         def check_response
-          if response.status >= 400
+          if response.status == 404
+            message = json_response.fetch_path("error", "message") || json_response["error"]
```
Comment (abellotti) [inline lib/manageiq/api/client/connection.rb]: I think this is fine, I just don't remember, when is json response populated as { "error" : "....." } vs. { "error" : { "kind" : "...", "message" : ".... " }, ...}

Comment (bdunne) [reply lib/manageiq/api/client/connection.rb]: I don't know, I was basing this on the existing examples.

**Comment by abellotti** [comment]
Thanks @bdunne for the enhancement. 👍

**Pull request merged**